File: GitHubPop/app/src/main/java/com/igormeira/githubpop/pullrequest/PullRequestsActivity.java
package com.igormeira.githubpop.pullrequest;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.ProgressDialog;
import android.os.Bundle;

import com.igormeira.githubpop.R;
import com.igormeira.githubpop.databinding.ActivityPullRequestsBinding;
import com.igormeira.githubpop.model.PullRequest;

import java.util.ArrayList;

import static com.igormeira.githubpop.handler.Dialog.showLoadingDialog;

public class PullRequestsActivity extends AppCompatActivity {

    private PullRequestViewModel pullRequestViewModel;
    private PullRequestRecyclerAdapter pullRequestRecyclerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityPullRequestsBinding activityPullRequestsBinding =
                DataBindingUtil.setContentView(this, R.layout.activity_pull_requests);

        // Bind RecyclerView
        RecyclerView recyclerView = activityPullRequestsBinding.recyclerViewPullRequest;
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setHasFixedSize(true);

        pullRequestViewModel = ViewModelProviders.of(this).get(PullRequestViewModel.class);
        pullRequestRecyclerAdapter = new PullRequestRecyclerAdapter();
        recyclerView.setAdapter(pullRequestRecyclerAdapter);
        getAllPullRequests();
        pullRequestRecyclerAdapter.setContext(this);
    }

    private void getAllPullRequests() {
        String creator = getIntent().getStringExtra("creator");
        String name = getIntent().getStringExtra("name");
        ProgressDialog dialog = showLoadingDialog(this);
        dialog.show();
        pullRequestViewModel.getAllPullRequests(creator, name).observe(this, pullRequests -> {
            pullRequestRecyclerAdapter.setPullRequestsList((ArrayList<PullRequest>) pullRequests);
            dialog.dismiss();
        });
    }
}

File: app/src/main/java/br/com/guilherme/solution/ui/pullRequests/PullRequestsActivity.kt
package br.com.guilherme.solution.ui.pullRequests

import android.app.AlertDialog
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import br.com.guilherme.solution.R
import br.com.guilherme.solution.component.DaggerActivityComponent
import br.com.guilherme.solution.models.PullRequest
import br.com.guilherme.solution.models.Repository
import br.com.guilherme.solution.module.ActivityModule
import br.com.guilherme.solution.util.EXTRA_REPO
import kotlinx.android.synthetic.main.activity_pull_requests.*
import kotlinx.android.synthetic.main.content_pull_requests.*
import javax.inject.Inject

class PullRequestsActivity : AppCompatActivity(), PullRequestsContract.View,
    PullRequestAdapter.onItemClickListener {

    @Inject
    lateinit var presenter: PullRequestsContract.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pull_requests)
        setSupportActionBar(toolbar)
        injectDependency()

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        presenter.attach(this)
        presenter.subscribe()
        initView()
    }

    override fun showErrorMessage(error: String) {
        val alert = AlertDialog.Builder(applicationContext)
        alert.setTitle(getString(R.string.atention))
        alert.setMessage(error)
        alert.setNeutralButton(
            getString(R.string.ok),
            { dialog, id ->
                dialog.dismiss()
            }
        )
        alert.create()

        alert.show()
    }

    override fun loadDataSuccess(pullRequests: List<PullRequest>) {
        val adapter = PullRequestAdapter(applicationContext, pullRequests.toMutableList(), this)
        recycler_view_pull_request!!.setLayoutManager(LinearLayoutManager(applicationContext))
        recycler_view_pull_request!!.setAdapter(adapter)
    }

    override fun itemDetail(pull: PullRequest) {
        val openURL = Intent(Intent.ACTION_VIEW, Uri.parse(pull.html_url))
        startActivity(openURL)
    }

    private fun injectDependency() {
        val activityComponent = DaggerActivityComponent.builder()
            .activityModule(ActivityModule(this))
            .build()

        activityComponent.inject(this)
    }

    private fun initView() {
        val repo = intent.getParcelableExtra<Repository>(EXTRA_REPO)
        presenter.loadData(repo!!.owner.login, repo.name)
    }
}
